examples/AspNetCore/Program.cs

```diff
@@ -14,67 +14,157 @@
+
+            builder.Services.Configure<ZipkinExporterOptions>(builder.Configuration.GetSection("Zipkin"));
+            break;
+
+        case "otlp":
+            options.AddOtlpExporter(otlpOptions =>
                 {
-                    builder.ClearProviders();
-                    builder.AddConsole();
-
-                    var logExporter = context.Configuration.GetValue<string>("UseLogExporter").ToLowerInvariant();
-                    switch (logExporter)
-                    {
-                        case "otlp":
-                            // Adding the OtlpExporter creates a GrpcChannel.
-                            // This switch must be set before creating a GrpcChannel when calling an insecure gRPC service.
-                            // See: https://docs.microsoft.com/aspnet/core/grpc/troubleshoot#call-insecure-grpc-services-with-net-core-client
-                            AppContext.SetSwitch("System.Net.Http.SocketsHttpHandler.Http2UnencryptedSupport", true);
-                            builder.AddOpenTelemetry(options =>
-                            {
-                                options.SetResourceBuilder(ResourceBuilder.CreateDefault().AddService(context.Configuration.GetValue<string>("Otlp:ServiceName")));
-                                options.AddOtlpExporter(otlpOptions =>
-                                 {
-                                     otlpOptions.Endpoint = new Uri(context.Configuration.GetValue<string>("Otlp:Endpoint"));
-                                 });
-                            });
-                            break;
-
-                        default:
-                            builder.AddOpenTelemetry(options =>
-                            {
-                                options.AddConsoleExporter();
-                            });
-                            break;
-                    }
-
-                    builder.Services.Configure<OpenTelemetryLoggerOptions>(opt =>
-                    {
-                        opt.IncludeScopes = true;
-                        opt.ParseStateValues = true;
-                        opt.IncludeFormattedMessage = true;
-                    });
+                    otlpOptions.Endpoint = new Uri(builder.Configuration.GetValue<string>("Otlp:Endpoint"));
                 });
+            break;
+
+        default:
+            options.AddConsoleExporter();
+
+            break;
     }
+});
+
+// For options which can be bound from IConfiguration.
+builder.Services.Configure<AspNetCoreInstrumentationOptions>(builder.Configuration.GetSection("AspNetCoreInstrumentation"));
+
+// Logging
+builder.Logging.ClearProviders();
+
+builder.Logging.AddOpenTelemetry(options =>
+{
+    options.SetResourceBuilder(resourceBuilder);
+    var logExporter = builder.Configuration.GetValue<string>("UseLogExporter").ToLowerInvariant();
+    switch (logExporter)
+    {
+        case "otlp":
+            options.AddOtlpExporter();
```

Comment:
this is missing endpoint reading from config file? its part of current example.

Reply:
End point reading is still there.